primitives/io/src/lib.rs

```diff
@@ -93,10 +93,22 @@ pub enum EcdsaVerifyError {
 pub enum KillStorageResult {
 	/// No key remains in the child trie.
 	AllRemoved(u32),
+
 	/// At least one key still resides in the child trie due to the supplied limit.
 	SomeRemaining(u32),
 }
 
+/// The outcome of calling `storage_kill`. Returned value is the number of storage items
+/// removed from the trie and also removed from overlay from making the `storage_kill` call.
+#[derive(PassByCodec, Encode, Decode)]
+pub enum KillStorageResultV2 {
```

Comment:
As the `version(2)` of `kill_storage` should now be local to this crate, you should not name this `V2` and instead rename the old one.

Reply:
And also make the old one non public

Reply:
Thanks for all the feedback @bkchr 
what do you think, I should rename the old one has?

Reply:
I have updated my PR, can you help review @bkchr @thiolliere